main/src/ca/uwaterloo/flix/language/phase/monomorph/Lowering.scala

```diff
@@ -88,6 +88,317 @@ object Lowering {
+    case LoweredAst.Def(sym, spec0, exp, loc) =>
+      val spec = lowerSpec(spec0)
+      val e = lowerExp(exp)
+      MonoAst.Def(sym, spec, e, loc)
+  }
+
+  /**
+    * Lowers the given enum `enum0`.
+    */
+  protected[monomorph] def lowerEnum(enum0: LoweredAst.Enum): MonoAst.Enum = enum0 match {
+    case LoweredAst.Enum(doc, ann, mod, sym, tparams0, _, cases0, loc) =>
+      val tparams = tparams0.map(lowerTypeParam)
+      val cases = cases0.map {
+        case (_, LoweredAst.Case(caseSym, tpes0, _, caseLoc)) =>
+          val tpes = tpes0.map(lowerType)
+          (caseSym, MonoAst.Case(caseSym, tpes, caseLoc))
+      }
+      MonoAst.Enum(doc, ann, mod, sym, tparams, cases, loc)
+  }
+
+  /**
+    * Lowers the given `effect`.
+    */
+  protected[monomorph] def lowerEffect(effect: LoweredAst.Effect): MonoAst.Effect = effect match {
+    case LoweredAst.Effect(doc, ann, mod, sym, ops0, loc) =>
+      // TODO EFFECT-TPARAMS use tparams
+      val ops = ops0.map(lowerOp)
+      MonoAst.Effect(doc, ann, mod, sym, ops, loc)
+  }
+
+  /**
+    * Lowers the given struct `struct0`.
+    */
+  protected[monomorph] def lowerStruct(struct0: LoweredAst.Struct): MonoAst.Struct = struct0 match {
+    case LoweredAst.Struct(doc, ann, mod, sym, tparams0, fields0, loc0) =>
+      val tparams = tparams0.map(lowerTypeParam)
+      val fields = fields0.map {
+        case LoweredAst.StructField(fieldSym, tpe, loc) => MonoAst.StructField(fieldSym, lowerType(tpe), loc)
+      }
+      MonoAst.Struct(doc, ann, mod, sym, tparams, fields, loc0)
+  }
+
+  /**
+    * Lowers the given `op`.
+    */
+  private def lowerOp(op: LoweredAst.Op): MonoAst.Op = op match {
+    case LoweredAst.Op(sym, spec0, loc) =>
+      val spec = lowerSpec(spec0)
+      MonoAst.Op(sym, spec, loc)
+  }
+
+  /**
+    * Lowers the given `spec0`.
+    */
+  private def lowerSpec(spec0: LoweredAst.Spec): MonoAst.Spec = spec0 match {
+    case LoweredAst.Spec(doc, ann, mod, _, fparams0, declaredScheme, retTpe, eff, _) =>
+      val fs = fparams0.map(lowerFormalParam)
+      val fType = lowerType(declaredScheme.base)
+      val rType = lowerType(retTpe)
+      MonoAst.Spec(doc, ann, mod, fs, fType, rType, eff, DefContext.Unknown)
+  }
+
+  /**
+    * Lowers `exp0` replacing all types with the lowered types and lowering channels and fixpoint to the primitives.
+    */
+  private def lowerExp(exp0: LoweredAst.Expr)(implicit ctx: Context, root: LoweredAst.Root, flix: Flix): MonoAst.Expr = exp0 match {
+    case LoweredAst.Expr.Cst(cst, tpe, loc) => MonoAst.Expr.Cst(cst, lowerType(tpe), loc)
+    case LoweredAst.Expr.Var(sym, tpe, loc) => MonoAst.Expr.Var(sym, lowerType(tpe), loc)
+    case LoweredAst.Expr.Lambda(fparam, exp, tpe, loc) =>
+      val p = lowerFormalParam(fparam)
+      val e = lowerExp(exp)
+      val t = lowerType(tpe)
+      MonoAst.Expr.Lambda(p, e, t, loc)
```

Comment:
lets add linebreaks between every case, so the code is easier to read. (Like already done elsewhere).